### ISSUE TITLE
doesn't work with laravel 5.3

### ISSUE THREAD

**Issue by yozman** [FriendsOfPHP/Sami#259]
could u update the require?
`illuminate/filesystem` require `symfony/finder 3.1.*`

**Comment by GrahamCampbell** [comment]
Already done on newest version.

**Comment by yozman** [comment]
@GrahamCampbell 
excuse me
`"symfony/filesystem": "~3.0"`
doesn't work with `laravel 5.3`

**Comment by GrahamCampbell** [comment]
How do you mean. ~3.0 means any 3.x, including 3.x.

**Comment by yozman** [comment]
`illuminate/filesystem` require `3.1.*`

**Comment by GrahamCampbell** [comment]
Yeh. 3.1.* is a subset of ~3.0.

**Comment by yozman** [comment]
@GrahamCampbell 
I already use `sami/sami: 3.3.*`
u can try it,
doesn't work with `laravel 5.3`

**Comment by GrahamCampbell** [comment]
That's because that old version of sami only work with symfony 2.

**Comment by yozman** [comment]
old version?
witch `sami/sami` should I use?
the newest release is `3.3.0`

**Comment by GrahamCampbell** [comment]
Proof this works:

```bash
nas10-240-235-227:test graham$ cat composer.json
{
    "name": "graham/test",
    "require": {
        "laravel/framework": "5.3.*",
        "sami/sami": "4.0.*"
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}

nas10-240-235-227:test graham$ composer install --prefer-dist --no-suggest
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 40 installs, 0 updates, 0 removals
  - Installing composer/ca-bundle (1.0.6) Loading from cache
  - Installing blackfire/php-sdk (v1.6.0) Loading from cache
  - Installing symfony/process (v3.1.8) Loading from cache
  - Installing symfony/yaml (v3.2.1) Loading from cache
  - Installing symfony/filesystem (v3.2.1) Loading from cache
  - Installing symfony/finder (v3.1.8) Loading from cache
  - Installing psr/log (1.0.2) Loading from cache
  - Installing symfony/debug (v3.1.8) Loading from cache
  - Installing symfony/polyfill-mbstring (v1.3.0) Loading from cache
  - Installing symfony/console (v3.1.8) Loading from cache
  - Installing nikic/php-parser (v3.0.2) Loading from cache
  - Installing twig/twig (2.x-dev ad0564a) Loading from cache
  - Installing pimple/pimple (v3.0.2) Loading from cache
  - Installing phpdocumentor/reflection-docblock (2.0.4) Loading from cache
  - Installing michelf/php-markdown (1.7.0) Loading from cache
  - Installing sami/sami (dev-master 2048fcb) Loading from cache
  - Installing vlucas/phpdotenv (v2.4.0) Loading from cache
  - Installing symfony/var-dumper (v3.1.8) Loading from cache
  - Installing symfony/translation (v3.1.8) Loading from cache
  - Installing symfony/routing (v3.1.8) Loading from cache
  - Installing symfony/http-foundation (v3.1.8) Loading from cache
  - Installing symfony/event-dispatcher (v3.2.1) Loading from cache
  - Installing symfony/http-kernel (v3.1.8) Loading from cache
  - Installing swiftmailer/swiftmailer (v5.4.5) Loading from cache
  - Installing paragonie/random_compat (v2.0.4) Loading from cache
  - Installing ramsey/uuid (3.5.2) Loading from cache
  - Installing jakub-onderka/php-console-color (0.1) Loading from cache
  - Installing jakub-onderka/php-console-highlighter (v0.3.2) Loading from cache
  - Installing dnoegel/php-xdg-base-dir (0.1) Loading from cache
  - Installing psy/psysh (v0.8.0) Loading from cache
  - Installing nesbot/carbon (1.21.0) Loading from cache
  - Installing mtdowling/cron-expression (v1.1.0) Loading from cache
  - Installing monolog/monolog (1.22.0) Loading from cache
  - Installing league/flysystem (1.0.32) Loading from cache
  - Installing symfony/polyfill-util (v1.3.0) Loading from cache
  - Installing symfony/polyfill-php56 (v1.3.0) Loading from cache
  - Installing jeremeamia/superclosure (2.3.0) Loading from cache
  - Installing doctrine/inflector (v1.1.0) Loading from cache
  - Installing classpreloader/classpreloader (3.1.0) Loading from cache
  - Installing laravel/framework (v5.3.28) Loading from cache
Writing lock file
Generating autoload files
```

**Comment by yozman** [comment]
@GrahamCampbell 
u mean I should use `"sami/sami": "dev-master"`?

**Comment by GrahamCampbell** [comment]
No, I mean you should use `^4.0` or `4.0.*`, like I did.

**Comment by GrahamCampbell** [comment]
As you can see, v3 does not support symfony 3: https://github.com/FriendsOfPHP/Sami/blob/v3.3.0/composer.json#L14-L27.

**Comment by GrahamCampbell** [comment]
Sami was never intended to be installed as a dependency anyway. It's odd that you'd want to do that.

Proof: https://github.com/FriendsOfPHP/Sami/blob/master/composer.json#L3.

**Comment by yozman** [comment]
@GrahamCampbell 
OK. got it, I'll try.
wish u release it asap.
I can't find it at packagist

**Comment by GrahamCampbell** [comment]
There is no release yet. It's not ready.

**Comment by fabpot** [comment]
Closing as it wll be fixed with 4.0. Anyway, I would not recommend anyone to add Sami as a dep in their composer.json, but use it as a standalone tool instead.
